WordPress/Classes/ViewRelated/Blog/My Site/MySiteViewController.swift

```diff
@@ -941,6 +943,22 @@ class MySiteViewController: UIViewController, NoResultsViewHost {
         self.blog = blog
     }
 
+    // MARK: - Blaze
+
+    private func getBlazeStatus(for blog: Blog?) {
+        guard FeatureFlag.blaze.enabled,
+              let blog = blog,
+              let blazeService = BlazeService() else {
+            return
+        }
+
+        blazeService.getStatus(for: blog, success: { approved in
+            print(approved)
```

Comment:
How about updating the menu if needed here?

Reply:
Also, maybe we can print a more informative log here. Or is will this be updated in a future PR?

Reply:
> How about updating the menu if needed here?

I decided to make the success closure optional and renamed the method to "updateBlazeStatus", since the way we handle setting the blog for BlogDetailsViewController is a bit different for (1) when we first set the blog vs (2) when we switch blogs.

> a more informative log here

Fixed! [b12a80e](https://github.com/wordpress-mobile/WordPress-iOS/pull/20127/commits/b12a80e976f776ac61d0fb521d2852bc23e72e1a)

Reply:
My understanding is that updating the blaze status is asynchronous. So theoretically, it could happen that we display the blaze menu item and then retrieve the blaze status, which returns false. In this scenario, we'll be displaying the menu item while we shouldn't until the menu is refreshed. The opposite scenario (not displaying the menu item while we should) is also possible. Am I missing something?